src/MCPClient/lib/clientScripts/verify_checksum.py

```diff
@@ -0,0 +1,276 @@
+        ext = os.path.splitext(path)[1]
+        if not ext:
+            return path
+        return ext.replace(".", "")
+
+    @staticmethod
+    def _count_lines(path):
+        """Count the number of lines in a checksum file."""
+        count = 0
+        with open(path) as hashfile:
+            for count, _ in enumerate(hashfile):
+                pass
+        # Negate zero-based count.
+        return count + 1
+
+    @staticmethod
+    def _count_files(path):
+        """Walk the directories on a given path and count the number of files.
+        """
+        return sum([len(files) for _, _, files in os.walk(path)])
+
+
+def write_premis_event_per_file(transfer_uuid, event_detail):
+    """Write a PREMIS event for every file in the transfer on success."""
+    group_type = "transfer_id"
+    event_type = "fixity_check"
+    event_outcome = "pass"
+    # Retrieve file UUIDs from the database.
+    kwargs = {"removedtime__isnull": True, group_type: transfer_uuid}
+    with transaction.atomic():
+        file_uuids = File.objects.filter(**kwargs).values_list("uuid")
+        if not file_uuids:
+            raise PREMISFailure()
+        for (fileUUID,) in file_uuids:
+            insertIntoEvents(
+                fileUUID=fileUUID,
+                eventType=event_type,
+                eventDateTime=datetime.datetime.now()
+                .replace(microsecond=0)
+                .isoformat(),
+                eventDetail=event_detail,
+                eventOutcome=event_outcome,
+                eventOutcomeDetailNote="",
+            )
+
+
+def run_hashsum_commands(job):
+    """Run hashsum commands and generate a cumulative return code."""
+    transfer_dir = None
+    transfer_uuid = None
+    try:
+        transfer_dir = job.args[1]
+        transfer_uuid = job.args[2]
+    except IndexError:
+        logger.error("Cannot access expected module arguments: %s", job.args)
+        return 1
+    ret = 0
+    for hashfile in Hashsum.COMMANDS:
+        os.chdir(transfer_dir)
+        objectsdir = "objects/"
+        hashsum = None
+        if os.path.exists(hashfile):
+            try:
+                hashsum = Hashsum(hashfile, job)
+            except NoHashCommandAvailable:
+                job.pyprint(
+                    "Nothing to do for {}. No command available.".format(
+                        Hashsum.get_ext(hashfile)
+                    )
+                )
+                continue
+        if hashsum:
+            job.pyprint(
+                "Comparing transfer checksums with the supplied {} file".format(
+                    Hashsum.get_ext(hashfile)
+                ),
+                file=sys.stderr,
+            )
+            result = hashsum.compare_hashes(hashfile=hashfile, objectsdir=objectsdir)
+            # Add to PREMIS on success only.
+            if result == 0:
+                job.pyprint("{}: Comparison was OK".format(Hashsum.get_ext(hashfile)))
```

Comment:
Should this be redirected to `file=sys.stderr` too?

Reply:
I'm not sure, my intent was to keep the output as user friendly/intuitive as possible. We'll end up with something along the lines of:
![image](https://user-images.githubusercontent.com/1880412/59462828-d1b86400-8e24-11e9-87e2-9c1673d88b29.png)